flink-docs/src/main/java/org/apache/flink/docs/configuration/ConfigOptionsDocGenerator.java

```diff
@@ -284,10 +290,31 @@ private static String toHtmlString(final OptionWithMetaInfo optionWithMetaInfo)
 			"        <tr>\n" +
 			"            <td><h5>" + escapeCharacters(option.key()) + "</h5>" + execModeStringBuilder.toString() + "</td>\n" +
 			"            <td style=\"word-wrap: break-word;\">" + escapeCharacters(addWordBreakOpportunities(defaultValue)) + "</td>\n" +
+			"            <td>" + type + "</td>\n" +
 			"            <td>" + formatter.format(option.description()) + "</td>\n" +
 			"        </tr>\n";
 	}
 
+	static String typeToHtml(OptionWithMetaInfo optionWithMetaInfo) {
+		ConfigOption<?> option = optionWithMetaInfo.option;
+		String typeName = option.getClazz().getSimpleName();
+		final String type;
+		if (option.isList()) {
+			type = String.format("List<%s>", typeName);
+		} else {
+			type = typeName;
+		}
+
+		if (option.getClazz().isEnum()) {
+			return String.format(
+				"<p>%s</p>Possible values: %s",
+				escapeCharacters(type),
```

Comment:
the actual class name is overkill imo and mostly an implementation detail; I'd just stick to `Enum: Possible values: ...`

Reply:
My only two cents on this one are that if in the future we want users to be able to specify these options also programatically, it would make sense to also have the name of the class.

But I am not so sure about the final result so I am also willing to leave it like this.

Reply:
I'm with @zentol on that one. I think for programmatic approach its enough that we have that type encoded in the `ConfigOption`. These docs are rather for defining options from config files.